codex/contracts/market.nim

```diff
@@ -36,7 +34,7 @@ proc approveFunds(market: OnChainMarket, amount: UInt256) {.async.} =
   let tokenAddress = await market.contract.token()
   let token = Erc20Token.new(tokenAddress, market.signer)
 
-  await token.approve(market.contract.address(), amount)
+  discard await token.approve(market.contract.address(), amount).confirm(1)
```

Comment:
I looked into this a bit more, but it doesn't seem that we can combine the `token.approve()` and a call into the marketplace into a single transaction. I was thinking of using [multicall](https://docs.openzeppelin.com/contracts/4.x/utilities#multicall), but that only works for batching calls to the same contract.

Other [sources](https://github.com/mds1/multicall#batch-contract-writes) seem to confirm that it is not possible to combine calls into different contracts, because it messes with `msg.sender`.